hivemind/moe/server/expert_backend.py

```diff
@@ -163,65 +141,47 @@ def backward(self, *inputs: torch.Tensor) -> Tuple[torch.Tensor, ...]:
             torch.autograd.backward(
                 outputs_flat, grad_tensors=grad_outputs_flat, create_graph=False, retain_graph=False
             )
-            self.apply_gradients(batch_size)
+            self.on_backward(batch_size)
```

Comment:
rationale: this does not necessarily apply gradients, e.g.
- virtual batching applies gradients once every k steps
- pretrained models do not apply the returned gradients